### DISCUSSION TITLE
GitHub Actions bot email address?

### DISCUSSION THREAD

**Discussion by paramt** [community/community#26560] [Actions]
<p>I wanted to commit to the repository from within a workflow. Simply running:</p>
<pre><code>git config --local user.email "action@github.com"
git config --local user.name "GitHub Action"
git commit -m "Add URL" -a
</code></pre>
<p>works, but GitHub action’s icon doesn’t show up. </p>
<p><img src="https://user-images.githubusercontent.com/23563074/181098462-4c86aefe-b858-4e5b-94a0-2e1da4d9b68e.png" alt="P4Q49mr" data-base62-sha1="mGVkqxF86xPRiK868ZoXzEQEeNR" width="229" height="26"></p>
<p>I couldn’t find the correct email address for the bot anywhere. Any help would be appreciated!</p>

**Comment by wipe2238** [comment]
<p>[nvm, readed words wrong]</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>This is an expected behavior. In commits author field, it will not show the black github-actions bot icon, just a gray one. It shows the name you specify in <strong>git config user.name</strong>.</p>
<p>Github Actions bot is just a bot account when you do operations with GITHUB_TOKEN. In pull requests, issues page, you can see its icon, but not in commits. Sorry for any inconvenience.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7fcb0abaa7e95c1f927c00de151cfd7e762a4729.png" data-download-href="/uploads/short-url/ievEy6GLSk7BCcIG5ZVYGb4MbBD.png?dl=1" title="github actions bot.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098465-5e0e6fdc-1642-40c5-99d8-b7ddff5301cc.png" alt="github actions bot.png" data-base62-sha1="ievEy6GLSk7BCcIG5ZVYGb4MbBD" width="690" height="281" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7fcb0abaa7e95c1f927c00de151cfd7e762a4729_2_690x281.png, https://aws1.discourse-cdn.com/github/original/2X/7/7fcb0abaa7e95c1f927c00de151cfd7e762a4729.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/7fcb0abaa7e95c1f927c00de151cfd7e762a4729.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7fcb0abaa7e95c1f927c00de151cfd7e762a4729_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">github actions bot.png</span><span class="informations">937×382 24.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/da3fa2e4370e2a83d21ab74fd5cd1c3468b71989.png" data-download-href="/uploads/short-url/v8IqzSZumwY9vyrhQNoRhvNrJIl.png?dl=1" title="issue bot.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181098466-eb0c0562-025f-4b33-8f16-fa86b8ab0b49.png" alt="issue bot.png" data-base62-sha1="v8IqzSZumwY9vyrhQNoRhvNrJIl" width="690" height="360" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/da3fa2e4370e2a83d21ab74fd5cd1c3468b71989_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">issue bot.png</span><span class="informations">707×369 21.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ianwalter** [comment]
<p>Would it be easy for GitHub to just assign an email address to the bot account so that commits with that email address would show the bot’s avatar?</p>

**Comment by ericnorris** [comment]
<p>I believe I was able to determine a valid email address to use by copying the format they use for dependabot:</p>
<p><a href="https://api.github.com/users/dependabot-preview%5Bbot%5D/events/public" rel="nofollow noopener">https://api.github.com/users/dependabot-preview[bot]/events/public</a></p>
<p>All of the commits are from </p>
<pre><code>27856297+dependabot-preview[bot]@users.noreply.github.com
</code></pre>
<p>…where 27856297 is the user ID for dependabot[bot].</p>
<p>So I tried:</p>
<pre><code>git commit --allow-empty --author="github-actions[bot] &lt;41898282+github-actions[bot]@users.noreply.github.com&gt;"
</code></pre>
<p>and it seemed to work:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/80c927108d408925da7b4c2890913b280e1ac4bd.png" data-download-href="/uploads/short-url/ini5bCiGf8cVyWoKDdhr6wd1CKV.png?dl=1" title="Screen Shot 2020-04-13 at 1.42.32 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/1906605/181098469-c4498e4c-ea58-4d7c-a47b-5e1215e834b9.png" alt="Screen Shot 2020-04-13 at 1.42.32 PM.png" data-base62-sha1="ini5bCiGf8cVyWoKDdhr6wd1CKV" width="690" height="151" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/80c927108d408925da7b4c2890913b280e1ac4bd_2_690x151.png, https://aws1.discourse-cdn.com/github/original/2X/8/80c927108d408925da7b4c2890913b280e1ac4bd.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/80c927108d408925da7b4c2890913b280e1ac4bd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/80c927108d408925da7b4c2890913b280e1ac4bd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-04-13 at 1.42.32 PM.png</span><span class="informations">1009×221 27.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Reply by voxpelli** [reply]
Trying to get the GitHub `actions/checkout` README to suggest this: https://github.com/actions/checkout/pull/1184

**Reply by vivodi** [reply]
`github-actions[bot]@users.noreply.github.com` is enough.

**Comment by weibeld** [comment]
<p>The email address suggested by <a class="mention" href="/ericnorris">@ericnorris</a> actually works:</p>
<pre><code class="lang-bash">41898282+github-actions[bot]@users.noreply.github.com
</code></pre>
<p>Using this for <code>git config user.email</code> results in:</p>
<p><img src="https://user-images.githubusercontent.com/2396208/181098472-4a57d703-c63a-4a96-98c2-9ffc3aae9ab8.png" alt="Screenshot 2020-07-23 at 14.36.08" data-base62-sha1="bLgg29c8FgtjRXMDXKVtBeN8HP5" width="504" height="63"></p>
<p><a href="https://api.github.com/users/github-actions%5Bbot%5D" rel="nofollow noopener">https://api.github.com/users/github-actions[bot]</a> seems to be the API entry for the <code>github-actions</code> bot, which also contains the ID 41898282.</p>

**Reply by vivodi** [reply]
`github-actions[bot]@users.noreply.github.com` is enough.

**Comment by qaisjp** [comment]
<p>Yep! This is amazing. <a class="mention" href="/weibeld">@weibeld</a> +rep to you my dude.</p>
<p>Just go to <a href="https://api.github.com/users/better-informatics%5Bbot%5D" rel="nofollow noopener">https://api.github.com/users/better-informatics[bot]</a> and use the ID from there :slight_smile:</p>

**Comment by umer2001** [comment]
When you want to **tag  commit  as GitHub Actions** then:

`git config user.email "41898282+github-actions[bot]@users.noreply.github.com"`
`git config user.name "github-actions[bot]"`

then `git tag -a -m "<msg>" <tag> <commit hash>`
**note**: `-a` is very important here
lastly `git push`

**Reply by caraar12345** [reply]
I came here looking for the GHA email so I can create tags with an action - and found this wonderfully helpful message. I wasn't totally sure on how `git tag` works so having confirmation that the command I've written is correct is stupidly useful. Thank you!

**Comment by qoomon** [comment]
I've created a handy action for that use case Feel free to have look at https://github.com/qoomon/actions--setup-git

**Reply by PatrickJS-kunaico** [reply]
thanks

**Comment by frenzyfriday** [comment]
I was looking for the email id of github-actions bot and that brought me here. The [readme](https://github.com/actions/checkout?tab=readme-ov-file#push-a-commit-to-a-pr-using-the-built-in-token) in actions/checkout points to how we can actually get the email id: https://api.github.com/users/github-actions%5Bbot%5D

**Comment by ALNAHER1** [comment]
> <p>[nvm, readed words wrong]</p>

- @thejoebourneidentity has 45

**Comment by jerrymathewdiscovery** [comment]
Any one have any details on how to generate keys using `gpg --full-generate-key ` to digitally sign commits using the 'github-action[bot]'


`459712 ~ % gpg --full-generate-key                                                  
gpg (GnuPG) 2.4.8; Copyright (C) 2025 g10 Code GmbH
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Please select what kind of key you want:
   (1) RSA and RSA
   (2) DSA and Elgamal
   (3) DSA (sign only)
   (4) RSA (sign only)
   (9) ECC (sign and encrypt) *default*
  (10) ECC (sign only)
  (14) Existing key from card
Your selection? 1
RSA keys may be between 1024 and 4096 bits long.
What keysize do you want? (3072) 4096
Requested keysize is 4096 bits
Please specify how long the key should be valid.
         0 = key does not expire
      <n>  = key expires in n days
      <n>w = key expires in n weeks
      <n>m = key expires in n months
      <n>y = key expires in n years
Key is valid for? (0) 0
Key does not expire at all
Is this correct? (y/N) y

GnuPG needs to construct a user ID to identify your key.

Real name: github-actions
Email address: 41898282+github-actions[bot]@users.noreply.github.com
Not a valid email address
Email address: 41898282+github-actions[bot]@users.noreply.github.com
Not a valid email address

`

**Reply by vivodi** [reply]
I think it's impossible since you don't own 41898282+github-actions[bot]@users.noreply.github.com

**Reply by qoomon** [reply]
Yes that's possible. You need to do it via api calls. Feel free to use my action for creating signed commit https://github.com/qoomon/actions--create-commit
